sale_promotion_rule/models/sale_order.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    promotion_rule_ids = fields.Many2many(
+        'sale.promotion.rule',
+        string='Promotion rules',
+        domain=[('rule_type', '!=', 'coupon')],
+        readonly=True
+    )
+
+    coupon_promotion_rule_id = fields.Many2one(
+        'sale.promotion.rule',
+        string='Coupon promotion rule',
+        domain=[('rule_type', '=', 'coupon')],
+        readonly=True
+    )
+    coupon_code = fields.Char(
+        related='coupon_promotion_rule_id.code',
+        readonly=True,
+        store=True
+    )
+
+    applied_promotion_rule_ids = fields.Many2many(
+        'sale.promotion.rule',
+        string='Promotion rules',
+        compute='_compute_applied_promotion_rule_ids'
```

Comment:
@lmignon IMO change `_compute_has_promotion_rules`

Reply:
@nikul-serpentcs IMO we must keep one compute method by field to ease readability, maintainability and possible extension. An all in one method is never a good idea.